src/cli/filter_fixtures.py

```diff
@@ -0,0 +1,434 @@
+"""
+Filter test fixtures by fork range.
+
+This tool filters test fixtures from an input directory to an output directory
+based on fork inclusion criteria, similar to the fill command's --until functionality.
+"""
+
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Type
+
+import click
+from rich.progress import track
+
+from ethereum_test_fixtures.file import Fixtures
+from ethereum_test_fixtures.pre_alloc_groups import PreAllocGroup
+from ethereum_test_forks import get_deployed_forks, get_fork_by_name
+from ethereum_test_forks.base_fork import BaseFork
+
+
+def get_valid_fork_range(
+    single_fork: str | None = None,
+    forks_from: str | None = None,
+    forks_until: str | None = None,
+) -> tuple[Type[BaseFork] | None, Type[BaseFork] | None, Type[BaseFork] | None]:
+    """
+    Get the fork range for filtering. Returns (single_fork, from_fork, until_fork).
+
+    This function validates the fork names but doesn't try to enumerate all valid forks,
+    since transition forks are dynamically created and not in the fork registry.
+    """
+    if single_fork:
+        fork = get_fork_by_name(single_fork)
+        if fork is None:
+            raise ValueError(f"Fork '{single_fork}' not found")
+        return fork, None, None
+
+    # Validate from fork if specified
+    from_fork = None
+    if forks_from:
+        from_fork = get_fork_by_name(forks_from)
+        if from_fork is None:
+            raise ValueError(f"Fork '{forks_from}' not found")
+
+    # Validate until fork if specified
+    until_fork = None
+    if forks_until:
+        until_fork = get_fork_by_name(forks_until)
+        if until_fork is None:
+            raise ValueError(f"Fork '{forks_until}' not found")
+
+    return None, from_fork, until_fork
+
+
+def should_include_fixture_fork(
+    fixture_fork: Type[BaseFork],
+    single_fork: Type[BaseFork] | None,
+    from_fork: Type[BaseFork] | None,
+    until_fork: Type[BaseFork] | None,
+) -> bool:
+    """
+    Determine if a fixture's fork should be included based on the filtering criteria.
+
+    This uses fork comparison operators to handle both regular and transition forks.
+    """
+    if single_fork:
+        return fixture_fork == single_fork
+
+    # Default range is all deployed forks if no range specified
+    if not from_fork and not until_fork:
+        deployed_forks = get_deployed_forks()
+        return deployed_forks[0] <= fixture_fork <= deployed_forks[-1]
+
+    # Handle range filtering
+    from_ok = True
+    until_ok = True
+
+    if from_fork:
+        from_ok = fixture_fork >= from_fork
+
+    if until_fork:
+        until_ok = fixture_fork <= until_fork
+
+    return from_ok and until_ok
```

Comment:
We should use the `get_selected_fork_set` function introduced in #2051 to match the exact logic that `fill` uses.

The result of that function is a set, so you can do `fork in get_selected_fork_set(...)`.